geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -51,106 +51,59 @@ object FilterHelper {
     * @param sft simple feature type
     * @return valid op
     */
-  def visitBinarySpatialOp(op: BinarySpatialOperator, sft: SimpleFeatureType): Filter = {
+  def visitBinarySpatialOp(op: BinarySpatialOperator, sft: SimpleFeatureType, factory: FilterFactory2): Filter = {
     val prop = org.locationtech.geomesa.filter.checkOrderUnsafe(op.getExpression1, op.getExpression2)
     val geom = prop.literal.evaluate(null, classOf[Geometry])
     if (geom.getUserData == SafeGeomString) {
       op // we've already visited this geom once
     } else {
       // check for null or empty attribute and replace with default geometry name
-      val attribute = Option(prop.name).filterNot(_.isEmpty).getOrElse(if (sft == null) null else sft.getGeomField)
+      val attribute = Option(prop.name).filterNot(_.isEmpty).orElse(Option(sft).map(_.getGeomField)).orNull
       // copy the geometry so we don't modify the original
-      val geomCopy = gf.createGeometry(geom)
+      val geomCopy = geom.getFactory.createGeometry(geom)
```

Comment:
hehehe.... Does Geometry have a clone method for this kind of use?

Reply:
hmm, it actually does have a clone() implementation - not sure why we didn't use that to begin with. this change was mainly just to preserve any precision models associated with the geom, probably irrelevant...